### ISSUE TITLE
ext_pillars eat other ext_pillar's file_roots

### ISSUE THREAD

**Issue by TronPaul** [saltstack/salt#26506]
Version: 2015.8

Given a configuration looking like:

```
#pillar_roots are unset

ext_pillar:
  - s3:
      bucket: pillar-production
      environment: production
  - git:
    - master https://github.com/User/pillar.git:
      - env: base
```

And a file tree looking like
`https://github.com/User/pillar.git`:

```
top.sls
a.sls
b.sls
```

`s3://pillar-production`:

```
a.sls
```

And a pillar top file like:

```
base:
  '*':
    - b
  '* and G@virtual:Virtualbox':
    - a
production:
  '*':
    - a
```

Salt will cache the `base` and `production` environment, but not include the `production` environment's cached path in `FileClient`'s `file_roots`. However if the configuration is changed to

```
pillar_roots:
  base:
    - /srv/pillar

ext_pillar:
  - s3:
      bucket: pillar-production
      environment: production
```

and the file trees are kept the same, Salt will find `a.sls` from the `production` environment. This also occurs if the environments are switched (define `production` in `pillar_roots` and base in `ext_pillar`). I have also tried switching the ext pillar types between git and s3.

**Comment by TronPaul** [comment]
The reason I mention eating file_roots, is that upon adding some debug logging the LocalClient claims it's roots are only the base.

Actually I've tried changing the order of the `ext_pillar`s and the file_root changes based on which `ext_pillar` definition is last. If `base` is last, only the `base`'s `file_root` is used. If `production` is last, only the `production`'s is used.

**Comment by jfindlay** [comment]
@TronPaul, thanks for the report.  @cachedout, a fix similar to #24878 may be what is needed here as well.

**Comment by DanyC97** [comment]
oh, i need this fix on 2015.8.1 too .. thank you!!
